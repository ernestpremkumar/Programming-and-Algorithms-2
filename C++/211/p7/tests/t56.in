breadth
balanced
size
echo 5
insert 5
distance
balanced
print
find 3
distance
breadth
breadth
breadth
find 3
breadth
balanced
echo 2
echo 6
find 2
balanced
find 4
balanced
distance
distance
find 1
distance
insert 0
print
distance
find 7
insert 4
distance
insert 1
find 6
balanced
insert 2
echo 2
breadth
print
balanced
balanced
echo 5
insert 0
distance
print
find 1
print
balanced
echo 5
insert 2
breadth
size
find 5
find 7
find 7
balanced
echo 2
find 0
print
echo 1
find 0
distance
distance
print
balanced
distance
echo 4
insert 0
balanced
size
balanced
breadth
size
echo 0
insert 6
balanced
print
find 5
size
insert 7
breadth
distance
balanced
echo 7
print
insert 2
print
size
distance
breadth
size
print
insert 7
insert 7
balanced
print
insert 4
insert 2
size
balanced
find 6
insert 0
distance
distance
breadth
find 3
balanced
distance
balanced
find 0
size
size
size
print
print
size
find 7
balanced
distance
breadth
insert 3
find 1
echo 1
insert 1
insert 7
breadth
breadth
echo 7
size
print
echo 6
distance
size
breadth
find 0
print
find 5
distance
echo 4
find 7
print
distance
echo 1
insert 6
size
insert 5
find 2
distance
breadth
distance
breadth
breadth
insert 1
breadth
find 1
find 0
insert 7
insert 1
balanced
breadth
balanced
print
distance
echo 2
breadth
find 6
echo 3
print
print
size
insert 7
balanced
balanced
breadth
size
insert 2
echo 1
find 5
echo 6
insert 3
insert 5
echo 5
distance
size
breadth
size
find 7
size
insert 2
insert 7
insert 3
print
echo 1
echo 7
find 0
breadth
distance
distance
size
balanced
breadth
balanced
size
find 1
echo 3
distance
balanced
size
insert 7
balanced
print
insert 4
size
balanced
balanced
breadth
balanced
breadth
breadth
print
breadth
distance
print
find 0
find 5
size
distance
insert 3
size
distance
distance
distance
balanced
distance
print
insert 2
print
breadth
balanced
distance
size
insert 0
find 1
size
find 5
distance
find 2
balanced
insert 2
distance
echo 5
balanced
distance
breadth
breadth
echo 3
find 7
echo 1
print
find 2
size
size
balanced
balanced
balanced
breadth
print
breadth
insert 2
echo 2
distance
breadth
insert 1
echo 6
echo 6
distance
breadth
echo 0
insert 4
print
breadth
insert 2
distance
print
echo 7
breadth
print
echo 3
insert 6
print
find 1
print
size
breadth
breadth
print
print
print
find 2
size
breadth
size
size
echo 2
size
breadth
distance
balanced
distance
find 4
distance
distance
breadth
breadth
find 1
breadth
distance
find 7
size
print
balanced
size
find 5
breadth
distance
breadth
echo 4
balanced
breadth
balanced
find 1
size
find 1
size
echo 5
size
echo 7
print
distance
breadth
insert 4
echo 2
breadth
echo 7
find 2
find 7
find 6
echo 5
size
print
insert 2
insert 6
size
breadth
distance
print
print
size
print
print
insert 0
distance
breadth
size
insert 7
size
print
print
insert 7
balanced
print
balanced
insert 1
size
find 5
breadth
distance
echo 4
size
size
insert 3
distance
echo 6
breadth
print
echo 4
echo 1
size
insert 1
balanced
find 5
insert 4
echo 1
print
breadth
distance
find 6
print
balanced
print
distance
print
print
echo 6
find 6
find 7
find 6
balanced
distance
balanced
insert 3
size
size
balanced
echo 1
print
echo 6
print
balanced
size
find 2
echo 5
size
find 7
size
print
balanced
echo 6
print
size
find 4
size
echo 3
echo 7
find 7
size
echo 7
distance
size
size
find 4
echo 6
print
print
insert 1
breadth
breadth
echo 0
balanced
breadth
size
insert 7
insert 2
print
echo 3
echo 7
breadth
distance
insert 2
find 2
echo 2
balanced
echo 6
find 7
balanced
size
balanced
print
find 1
print
balanced
balanced
breadth
balanced
balanced
insert 2
size
breadth
balanced
size
insert 6
balanced
echo 1
size
print
distance
find 3
echo 5
size
distance
insert 1
insert 7
balanced
balanced
balanced
print
distance
size
balanced
insert 4
size
balanced
breadth
breadth
balanced
insert 4
find 0
print
echo 3
insert 3
breadth
insert 6
size
insert 3
balanced
breadth
insert 1
balanced
find 6
distance
echo 2
find 7
insert 4
balanced
find 1
insert 0
distance
find 2
breadth
size
distance
distance
find 7
distance
insert 6
insert 4
find 2
distance
find 4
breadth
breadth
distance
insert 4
print
breadth
find 6
breadth
find 1
distance
balanced
breadth
size
print
print
breadth
distance
echo 2
distance
distance
insert 3
insert 3
balanced
distance
balanced
find 4
print
distance
echo 6
distance
size
breadth
distance
print
echo 4
balanced
distance
breadth
echo 5
balanced
echo 7
distance
size
find 4
balanced
distance
breadth
insert 3
insert 3
balanced
echo 1
size
echo 6
find 0
find 0
balanced
print
echo 5
size
echo 3
print
size
insert 1
print
echo 3
print
find 0
distance
balanced
find 1
find 2
distance
echo 7
breadth
insert 2
find 5
find 5
balanced
echo 5
echo 3
balanced
insert 4
find 2
breadth
insert 2
breadth
find 4
print
insert 3
breadth
balanced
balanced
breadth
balanced
size
breadth
print
print
distance
size
size
find 7
size
echo 7
insert 3
breadth
breadth
size
find 2
balanced
insert 5
distance
find 5
breadth
print
echo 7
distance
find 3
find 0
find 3
echo 4
print
print
balanced
echo 7
print
size
balanced
breadth
distance
find 1
balanced
size
insert 2
breadth
insert 4
print
size
find 1
echo 1
balanced
distance
echo 2
balanced
size
balanced
print
balanced
insert 6
find 0
breadth
print
insert 1
size
echo 5
insert 3
echo 7
insert 7
print
breadth
balanced
print
find 3
size
find 4
size
size
size
breadth
echo 4
balanced
print
echo 7
balanced
balanced
find 5
insert 6
insert 7
size
echo 3
insert 6
distance
insert 3
distance
breadth
find 7
echo 4
print
distance
distance
echo 5
find 1
insert 2
distance
echo 3
insert 7
size
print
echo 3
size
insert 5
size
breadth
find 6
insert 3
echo 1
size
print
echo 3
balanced
insert 0
insert 0
echo 3
balanced
balanced
distance
distance
find 0
distance
balanced
balanced
size
find 1
breadth
echo 3
find 4
print
breadth
insert 5
distance
breadth
size
breadth
echo 5
find 3
find 7
balanced
find 5
find 3
distance
print
distance
distance
distance
echo 0
insert 4
echo 5
insert 3
echo 5
insert 1
insert 3
breadth
size
find 0
insert 5
insert 2
print
distance
find 2
size
echo 3
echo 3
distance
distance
find 6
insert 7
print
breadth
balanced
find 0
size
breadth
print
insert 4
echo 7
size
balanced
size
breadth
print
echo 0
breadth
print
size
size
find 5
size
find 0
print
distance
echo 5
distance
print
print
echo 1
insert 0
distance
print
echo 2
breadth
print
echo 0
echo 6
breadth
echo 5
size
breadth
balanced
echo 1
size
print
insert 3
size
print
print
breadth
insert 0
distance
insert 1
distance
echo 1
insert 0
echo 5
distance
find 0
insert 0
find 7
insert 2
balanced
echo 5
distance
find 5
find 7
insert 2
distance
breadth
echo 6
find 6
distance
insert 1
insert 1
insert 2
print
size
echo 3
balanced
balanced
distance
print
distance
distance
echo 3
insert 6
balanced
print
insert 0
size
size
print
insert 4
distance
find 0
find 5
echo 1
echo 7
size
size
breadth
distance
find 4
print
echo 4
distance
print
distance
balanced
distance
insert 2
find 1
echo 1
echo 3
insert 2
breadth
breadth
size
breadth
print
print
balanced
breadth
distance
insert 4
distance
breadth
print
print
echo 4
distance
print
echo 4
print
breadth
find 1
find 1
size
echo 0
print
print
print
find 1
distance
breadth
breadth
find 4
echo 1
balanced
insert 4
breadth
distance
distance
insert 0
echo 6
echo 6
echo 4
breadth
breadth
find 3
size
balanced
insert 5
insert 7
find 1
breadth
distance
size
find 4
find 3
insert 3
balanced
balanced
distance
echo 5
echo 1
size
balanced
distance
print
size
size
breadth
breadth
echo 1
print
find 1
find 4
echo 2
balanced
find 0
size
distance
