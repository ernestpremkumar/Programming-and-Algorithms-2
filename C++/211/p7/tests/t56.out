{}
Tree is balanced.
0
5
Average distance of nodes to root = 0
Tree is balanced.
{5}
<3> is not in tree.
Average distance of nodes to root = 0
{5}
{5}
{5}
<3> is not in tree.
{5}
Tree is balanced.
2
6
<2> is not in tree.
Tree is balanced.
<4> is not in tree.
Tree is balanced.
Average distance of nodes to root = 0
Average distance of nodes to root = 0
<1> is not in tree.
Average distance of nodes to root = 0
{0, 5}
Average distance of nodes to root = 0.5
<7> is not in tree.
Average distance of nodes to root = 1
<6> is not in tree.
Tree is not balanced.
2
{5, 0, 4, 1, 2}
{0, 1, 2, 4, 5}
Tree is not balanced.
Tree is not balanced.
5
Average distance of nodes to root = 2
{0, 1, 2, 4, 5}
<1> is in tree.
{0, 1, 2, 4, 5}
Tree is not balanced.
5
{5, 0, 4, 1, 2}
5
<5> is in tree.
<7> is not in tree.
<7> is not in tree.
Tree is not balanced.
2
<0> is in tree.
{0, 1, 2, 4, 5}
1
<0> is in tree.
Average distance of nodes to root = 2
Average distance of nodes to root = 2
{0, 1, 2, 4, 5}
Tree is not balanced.
Average distance of nodes to root = 2
4
Tree is not balanced.
5
Tree is not balanced.
{5, 0, 4, 1, 2}
5
0
Tree is not balanced.
{0, 1, 2, 4, 5, 6}
<5> is in tree.
6
{5, 0, 6, 4, 7, 1, 2}
Average distance of nodes to root = 1.85714
Tree is not balanced.
7
{0, 1, 2, 4, 5, 6, 7}
{0, 1, 2, 4, 5, 6, 7}
7
Average distance of nodes to root = 1.85714
{5, 0, 6, 4, 7, 1, 2}
7
{0, 1, 2, 4, 5, 6, 7}
Tree is not balanced.
{0, 1, 2, 4, 5, 6, 7}
7
Tree is not balanced.
<6> is in tree.
Average distance of nodes to root = 1.85714
Average distance of nodes to root = 1.85714
{5, 0, 6, 4, 7, 1, 2}
<3> is not in tree.
Tree is not balanced.
Average distance of nodes to root = 1.85714
Tree is not balanced.
<0> is in tree.
7
7
7
{0, 1, 2, 4, 5, 6, 7}
{0, 1, 2, 4, 5, 6, 7}
7
<7> is in tree.
Tree is not balanced.
Average distance of nodes to root = 1.85714
{5, 0, 6, 4, 7, 1, 2}
<1> is in tree.
1
{5, 0, 6, 4, 7, 1, 2, 3}
{5, 0, 6, 4, 7, 1, 2, 3}
7
8
{0, 1, 2, 3, 4, 5, 6, 7}
6
Average distance of nodes to root = 2.25
8
{5, 0, 6, 4, 7, 1, 2, 3}
<0> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
<5> is in tree.
Average distance of nodes to root = 2.25
4
<7> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
1
8
<2> is in tree.
Average distance of nodes to root = 2.25
{5, 0, 6, 4, 7, 1, 2, 3}
Average distance of nodes to root = 2.25
{5, 0, 6, 4, 7, 1, 2, 3}
{5, 0, 6, 4, 7, 1, 2, 3}
{5, 0, 6, 4, 7, 1, 2, 3}
<1> is in tree.
<0> is in tree.
Tree is not balanced.
{5, 0, 6, 4, 7, 1, 2, 3}
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
2
{5, 0, 6, 4, 7, 1, 2, 3}
<6> is in tree.
3
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
8
Tree is not balanced.
Tree is not balanced.
{5, 0, 6, 4, 7, 1, 2, 3}
8
1
<5> is in tree.
6
5
Average distance of nodes to root = 2.25
8
{5, 0, 6, 4, 7, 1, 2, 3}
8
<7> is in tree.
8
{0, 1, 2, 3, 4, 5, 6, 7}
1
7
<0> is in tree.
{5, 0, 6, 4, 7, 1, 2, 3}
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
8
Tree is not balanced.
{5, 0, 6, 4, 7, 1, 2, 3}
Tree is not balanced.
8
<1> is in tree.
3
Average distance of nodes to root = 2.25
Tree is not balanced.
8
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
8
Tree is not balanced.
Tree is not balanced.
{5, 0, 6, 4, 7, 1, 2, 3}
Tree is not balanced.
{5, 0, 6, 4, 7, 1, 2, 3}
{5, 0, 6, 4, 7, 1, 2, 3}
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 0, 6, 4, 7, 1, 2, 3}
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
<0> is in tree.
<5> is in tree.
8
Average distance of nodes to root = 2.25
8
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
Tree is not balanced.
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 0, 6, 4, 7, 1, 2, 3}
Tree is not balanced.
Average distance of nodes to root = 2.25
8
<1> is in tree.
8
<5> is in tree.
Average distance of nodes to root = 2.25
<2> is in tree.
Tree is not balanced.
Average distance of nodes to root = 2.25
5
Tree is not balanced.
Average distance of nodes to root = 2.25
{5, 0, 6, 4, 7, 1, 2, 3}
{5, 0, 6, 4, 7, 1, 2, 3}
3
<7> is in tree.
1
{0, 1, 2, 3, 4, 5, 6, 7}
<2> is in tree.
8
8
Tree is not balanced.
Tree is not balanced.
Tree is not balanced.
{5, 0, 6, 4, 7, 1, 2, 3}
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 0, 6, 4, 7, 1, 2, 3}
2
Average distance of nodes to root = 2.25
{5, 0, 6, 4, 7, 1, 2, 3}
6
6
Average distance of nodes to root = 2.25
{5, 0, 6, 4, 7, 1, 2, 3}
0
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 0, 6, 4, 7, 1, 2, 3}
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
7
{5, 0, 6, 4, 7, 1, 2, 3}
{0, 1, 2, 3, 4, 5, 6, 7}
3
{0, 1, 2, 3, 4, 5, 6, 7}
<1> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
8
{5, 0, 6, 4, 7, 1, 2, 3}
{5, 0, 6, 4, 7, 1, 2, 3}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
<2> is in tree.
8
{5, 0, 6, 4, 7, 1, 2, 3}
8
8
2
8
{5, 0, 6, 4, 7, 1, 2, 3}
Average distance of nodes to root = 2.25
Tree is not balanced.
Average distance of nodes to root = 2.25
<4> is in tree.
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
{5, 0, 6, 4, 7, 1, 2, 3}
{5, 0, 6, 4, 7, 1, 2, 3}
<1> is in tree.
{5, 0, 6, 4, 7, 1, 2, 3}
Average distance of nodes to root = 2.25
<7> is in tree.
8
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
8
<5> is in tree.
{5, 0, 6, 4, 7, 1, 2, 3}
Average distance of nodes to root = 2.25
{5, 0, 6, 4, 7, 1, 2, 3}
4
Tree is not balanced.
{5, 0, 6, 4, 7, 1, 2, 3}
Tree is not balanced.
<1> is in tree.
8
<1> is in tree.
8
5
8
7
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
{5, 0, 6, 4, 7, 1, 2, 3}
2
{5, 0, 6, 4, 7, 1, 2, 3}
7
<2> is in tree.
<7> is in tree.
<6> is in tree.
5
8
{0, 1, 2, 3, 4, 5, 6, 7}
8
{5, 0, 6, 4, 7, 1, 2, 3}
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
8
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
{5, 0, 6, 4, 7, 1, 2, 3}
8
8
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
8
<5> is in tree.
{5, 0, 6, 4, 7, 1, 2, 3}
Average distance of nodes to root = 2.25
4
8
8
Average distance of nodes to root = 2.25
6
{5, 0, 6, 4, 7, 1, 2, 3}
{0, 1, 2, 3, 4, 5, 6, 7}
4
1
8
Tree is not balanced.
<5> is in tree.
1
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 0, 6, 4, 7, 1, 2, 3}
Average distance of nodes to root = 2.25
<6> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
6
<6> is in tree.
<7> is in tree.
<6> is in tree.
Tree is not balanced.
Average distance of nodes to root = 2.25
Tree is not balanced.
8
8
Tree is not balanced.
1
{0, 1, 2, 3, 4, 5, 6, 7}
6
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
8
<2> is in tree.
5
8
<7> is in tree.
8
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
6
{0, 1, 2, 3, 4, 5, 6, 7}
8
<4> is in tree.
8
3
7
<7> is in tree.
8
7
Average distance of nodes to root = 2.25
8
8
<4> is in tree.
6
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 0, 6, 4, 7, 1, 2, 3}
{5, 0, 6, 4, 7, 1, 2, 3}
0
Tree is not balanced.
{5, 0, 6, 4, 7, 1, 2, 3}
8
{0, 1, 2, 3, 4, 5, 6, 7}
3
7
{5, 0, 6, 4, 7, 1, 2, 3}
Average distance of nodes to root = 2.25
<2> is in tree.
2
Tree is not balanced.
6
<7> is in tree.
Tree is not balanced.
8
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
<1> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Tree is not balanced.
{5, 0, 6, 4, 7, 1, 2, 3}
Tree is not balanced.
Tree is not balanced.
8
{5, 0, 6, 4, 7, 1, 2, 3}
Tree is not balanced.
8
Tree is not balanced.
1
8
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
<3> is in tree.
5
8
Average distance of nodes to root = 2.25
Tree is not balanced.
Tree is not balanced.
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
8
Tree is not balanced.
8
Tree is not balanced.
{5, 0, 6, 4, 7, 1, 2, 3}
{5, 0, 6, 4, 7, 1, 2, 3}
Tree is not balanced.
<0> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
3
{5, 0, 6, 4, 7, 1, 2, 3}
8
Tree is not balanced.
{5, 0, 6, 4, 7, 1, 2, 3}
Tree is not balanced.
<6> is in tree.
Average distance of nodes to root = 2.25
2
<7> is in tree.
Tree is not balanced.
<1> is in tree.
Average distance of nodes to root = 2.25
<2> is in tree.
{5, 0, 6, 4, 7, 1, 2, 3}
8
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
<7> is in tree.
Average distance of nodes to root = 2.25
<2> is in tree.
Average distance of nodes to root = 2.25
<4> is in tree.
{5, 0, 6, 4, 7, 1, 2, 3}
{5, 0, 6, 4, 7, 1, 2, 3}
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 0, 6, 4, 7, 1, 2, 3}
<6> is in tree.
{5, 0, 6, 4, 7, 1, 2, 3}
<1> is in tree.
Average distance of nodes to root = 2.25
Tree is not balanced.
{5, 0, 6, 4, 7, 1, 2, 3}
8
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 0, 6, 4, 7, 1, 2, 3}
Average distance of nodes to root = 2.25
2
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
Tree is not balanced.
Average distance of nodes to root = 2.25
Tree is not balanced.
<4> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
6
Average distance of nodes to root = 2.25
8
{5, 0, 6, 4, 7, 1, 2, 3}
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
4
Tree is not balanced.
Average distance of nodes to root = 2.25
{5, 0, 6, 4, 7, 1, 2, 3}
5
Tree is not balanced.
7
Average distance of nodes to root = 2.25
8
<4> is in tree.
Tree is not balanced.
Average distance of nodes to root = 2.25
{5, 0, 6, 4, 7, 1, 2, 3}
Tree is not balanced.
1
8
6
<0> is in tree.
<0> is in tree.
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
5
8
3
{0, 1, 2, 3, 4, 5, 6, 7}
8
{0, 1, 2, 3, 4, 5, 6, 7}
3
{0, 1, 2, 3, 4, 5, 6, 7}
<0> is in tree.
Average distance of nodes to root = 2.25
Tree is not balanced.
<1> is in tree.
<2> is in tree.
Average distance of nodes to root = 2.25
7
{5, 0, 6, 4, 7, 1, 2, 3}
<5> is in tree.
<5> is in tree.
Tree is not balanced.
5
3
Tree is not balanced.
<2> is in tree.
{5, 0, 6, 4, 7, 1, 2, 3}
{5, 0, 6, 4, 7, 1, 2, 3}
<4> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 0, 6, 4, 7, 1, 2, 3}
Tree is not balanced.
Tree is not balanced.
{5, 0, 6, 4, 7, 1, 2, 3}
Tree is not balanced.
8
{5, 0, 6, 4, 7, 1, 2, 3}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
8
8
<7> is in tree.
8
7
{5, 0, 6, 4, 7, 1, 2, 3}
{5, 0, 6, 4, 7, 1, 2, 3}
8
<2> is in tree.
Tree is not balanced.
Average distance of nodes to root = 2.25
<5> is in tree.
{5, 0, 6, 4, 7, 1, 2, 3}
{0, 1, 2, 3, 4, 5, 6, 7}
7
Average distance of nodes to root = 2.25
<3> is in tree.
<0> is in tree.
<3> is in tree.
4
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
7
{0, 1, 2, 3, 4, 5, 6, 7}
8
Tree is not balanced.
{5, 0, 6, 4, 7, 1, 2, 3}
Average distance of nodes to root = 2.25
<1> is in tree.
Tree is not balanced.
8
{5, 0, 6, 4, 7, 1, 2, 3}
{0, 1, 2, 3, 4, 5, 6, 7}
8
<1> is in tree.
1
Tree is not balanced.
Average distance of nodes to root = 2.25
2
Tree is not balanced.
8
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
<0> is in tree.
{5, 0, 6, 4, 7, 1, 2, 3}
{0, 1, 2, 3, 4, 5, 6, 7}
8
5
7
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 0, 6, 4, 7, 1, 2, 3}
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
<3> is in tree.
8
<4> is in tree.
8
8
8
{5, 0, 6, 4, 7, 1, 2, 3}
4
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
7
Tree is not balanced.
Tree is not balanced.
<5> is in tree.
8
3
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
{5, 0, 6, 4, 7, 1, 2, 3}
<7> is in tree.
4
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
5
<1> is in tree.
Average distance of nodes to root = 2.25
3
8
{0, 1, 2, 3, 4, 5, 6, 7}
3
8
8
{5, 0, 6, 4, 7, 1, 2, 3}
<6> is in tree.
1
8
{0, 1, 2, 3, 4, 5, 6, 7}
3
Tree is not balanced.
3
Tree is not balanced.
Tree is not balanced.
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
<0> is in tree.
Average distance of nodes to root = 2.25
Tree is not balanced.
Tree is not balanced.
8
<1> is in tree.
{5, 0, 6, 4, 7, 1, 2, 3}
3
<4> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 0, 6, 4, 7, 1, 2, 3}
Average distance of nodes to root = 2.25
{5, 0, 6, 4, 7, 1, 2, 3}
8
{5, 0, 6, 4, 7, 1, 2, 3}
5
<3> is in tree.
<7> is in tree.
Tree is not balanced.
<5> is in tree.
<3> is in tree.
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
0
5
5
{5, 0, 6, 4, 7, 1, 2, 3}
8
<0> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
<2> is in tree.
8
3
3
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
<6> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 0, 6, 4, 7, 1, 2, 3}
Tree is not balanced.
<0> is in tree.
8
{5, 0, 6, 4, 7, 1, 2, 3}
{0, 1, 2, 3, 4, 5, 6, 7}
7
8
Tree is not balanced.
8
{5, 0, 6, 4, 7, 1, 2, 3}
{0, 1, 2, 3, 4, 5, 6, 7}
0
{5, 0, 6, 4, 7, 1, 2, 3}
{0, 1, 2, 3, 4, 5, 6, 7}
8
8
<5> is in tree.
8
<0> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
5
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
1
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
2
{5, 0, 6, 4, 7, 1, 2, 3}
{0, 1, 2, 3, 4, 5, 6, 7}
0
6
{5, 0, 6, 4, 7, 1, 2, 3}
5
8
{5, 0, 6, 4, 7, 1, 2, 3}
Tree is not balanced.
1
8
{0, 1, 2, 3, 4, 5, 6, 7}
8
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 0, 6, 4, 7, 1, 2, 3}
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
1
5
Average distance of nodes to root = 2.25
<0> is in tree.
<7> is in tree.
Tree is not balanced.
5
Average distance of nodes to root = 2.25
<5> is in tree.
<7> is in tree.
Average distance of nodes to root = 2.25
{5, 0, 6, 4, 7, 1, 2, 3}
6
<6> is in tree.
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
8
3
Tree is not balanced.
Tree is not balanced.
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
3
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
8
8
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
<0> is in tree.
<5> is in tree.
1
7
8
8
{5, 0, 6, 4, 7, 1, 2, 3}
Average distance of nodes to root = 2.25
<4> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
4
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
Tree is not balanced.
Average distance of nodes to root = 2.25
<1> is in tree.
1
3
{5, 0, 6, 4, 7, 1, 2, 3}
{5, 0, 6, 4, 7, 1, 2, 3}
8
{5, 0, 6, 4, 7, 1, 2, 3}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
{5, 0, 6, 4, 7, 1, 2, 3}
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
{5, 0, 6, 4, 7, 1, 2, 3}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
4
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
4
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 0, 6, 4, 7, 1, 2, 3}
<1> is in tree.
<1> is in tree.
8
0
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
<1> is in tree.
Average distance of nodes to root = 2.25
{5, 0, 6, 4, 7, 1, 2, 3}
{5, 0, 6, 4, 7, 1, 2, 3}
<4> is in tree.
1
Tree is not balanced.
{5, 0, 6, 4, 7, 1, 2, 3}
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
6
6
4
{5, 0, 6, 4, 7, 1, 2, 3}
{5, 0, 6, 4, 7, 1, 2, 3}
<3> is in tree.
8
Tree is not balanced.
<1> is in tree.
{5, 0, 6, 4, 7, 1, 2, 3}
Average distance of nodes to root = 2.25
8
<4> is in tree.
<3> is in tree.
Tree is not balanced.
Tree is not balanced.
Average distance of nodes to root = 2.25
5
1
8
Tree is not balanced.
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
8
8
{5, 0, 6, 4, 7, 1, 2, 3}
{5, 0, 6, 4, 7, 1, 2, 3}
1
{0, 1, 2, 3, 4, 5, 6, 7}
<1> is in tree.
<4> is in tree.
2
Tree is not balanced.
<0> is in tree.
8
Average distance of nodes to root = 2.25
